### PR TITLE
Adjust functions in wc-formatting-functions.php to handle null argument deprecations in PHP 8.1

### LINKED ISSUE #36528: wc_format_decimal and null (Deprecated warning in PHP 8)
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

While using WooCommerce Product CRUD, there are many calls to "wc_format_decimal()". There is a line with a str_replace that says the following:

$number = str_replace( $decimals, '.', $number );

If $number is empty string (eg ''), it doesn't product any warning. If $number is null, it passed the !float if and the following error occurs:
Deprecated: str_replace(): Passing null to parameter #3 ($subject) of type array|string is deprecated

Possible solution:
Change if statement to check for string/array 
if ( ! is_float( $number )  && is_string( $number ))

To this line
https://github.com/woocommerce/woocommerce/blob/7d8f7ad57c99fe1653b474bcc944eaf1b7e830b4/plugins/woocommerce/includes/wc-formatting-functions.php#L295

### Expected behavior

Prevent str_replace run on non string/array subject

### Actual behavior

Deprecated: str_replace(): Passing null to parameter #3 ($subject) of type array|string is deprecated

### Steps to reproduce

1) Create a product with WooCommerce CRUD
```
$product=new WC_Product();
$product->set_regular_price("10");
$product->set_sale_price(null);
$product->save();

```

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by Konamiman**
- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

In PHP 8.1 several functions that accept a string as an argument (e.g. `str_replace`) throw a deprecation notice when null is passed. This commit changes the usages of these functions in `wc-formatting-functions.php` so that null inputs are converted to empty strings.

Closes #36528.

### How to test the changes in this Pull Request:

Switch to PHP 8.1 and run the following fom command line:

```
wp eval 'error_reporting(E_ALL); var_dump(wc_string_to_bool(null));'
```

Without this fix you'll see "Passing null to parameter #1 ($string) of type string is deprecated" or a similar error. With this fix you won't see any error.

You can repeat with all the other functions from `wc-formatting-functions.php` that have been adjusted.

### CODE REVIEW

**Comment by jorgeatorres** [comment]
@Konamiman: I don't see any references to `wc-page-functions.php` in this PR. Is something missing or does the title need to be adjusted?

File: plugins/woocommerce/changelog/fix-php-8.1-null-args-deprecation [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: dev
+
+Adjust wc-page-functions.php null args deprecations in PHP 8.1
```
Comment (barryhughes) [inline plugins/woocommerce/changelog/fix-php-8.1-null-args-deprecation]: Will wait on an answer to [Jorge's question](https://github.com/woocommerce/woocommerce/pull/38635#issuecomment-1581012757) (might need a change here).

**Comment by Konamiman** [comment]
@jorgeatorres @barryhughes My brain must have been especially off base yesterday. Yes, the edited file is `wc-formatting-functions.php` and I have no idea where that `wc-page-functions.php` came from. I have edited the PR description, the changelog file, and the commit message.

**Review by barryhughes — APPROVED** [review]
LGTM 💪🏼 

Tested a number of funcs besides `wc_string_to_bool()`, working well.

File: plugins/woocommerce/includes/wc-formatting-functions.php [hunk 2]
```diff
@@ -13,6 +13,8 @@
 
 defined( 'ABSPATH' ) || exit;
 
+//TODO: Once WooCommerce requires PHP 7.4, the "$x = $x ?? ''" constructs can be replaced with "$x ??= ''".
```
Comment (barryhughes) [inline plugins/woocommerce/includes/wc-formatting-functions.php]: Couple of PHPCS violations here:

```
WARNING | Comment refers to a TODO task (Generic.Commenting.Todo.TaskFound)
ERROR | No space found before comment text (Squiz.Commenting.InlineComment.NoSpaceBefore)
```

I'm assuming we only need to clear the error and not the warning:

```suggestion
// TODO: Once WooCommerce requires PHP 7.4, the "$x = $x ?? ''" constructs can be replaced with "$x ??= ''".
```

Comment (barryhughes) [reply plugins/woocommerce/includes/wc-formatting-functions.php]: Committed the suggestion, will see if that unblocks merge.

File: plugins/woocommerce/includes/wc-formatting-functions.php [hunk 3]
```diff
@@ -13,6 +13,8 @@
 
 defined( 'ABSPATH' ) || exit;
 
+// TODO: Once WooCommerce requires PHP 7.4, the "$x = $x ?? ''" constructs can be replaced with "$x ??= ''".
```
Comment (barryhughes) [inline plugins/woocommerce/includes/wc-formatting-functions.php]:
```suggestion
// Once WooCommerce requires PHP 7.4, the "$x = $x ?? ''" constructs can be replaced with "$x ??= ''".
```

**Pull request merged**